src/tools/illink/src/linker/Linker/AssemblyResolver.cs

```diff
@@ -47,16 +47,22 @@ public class AssemblyResolver : IAssemblyResolver
 		readonly Dictionary<string, AssemblyDefinition> _pathToAssembly = new ();
 		readonly List<MemoryMappedViewStream> _viewStreams = new ();
 		readonly ReaderParameters _defaultReaderParameters;
+		readonly string[] _extensions;
 
 		HashSet<string>? _unresolvedAssemblies;
 		HashSet<string>? _reportedUnresolvedAssemblies;
 
 		public AssemblyResolver (LinkContext context)
+			: this(context, new ReaderParameters (), Extensions)
 		{
+		}
+
+		protected AssemblyResolver (LinkContext context, ReaderParameters readerParameters, string[] extensions)
+		{
```

Comment:
Would it make sense to modify the public constructor by either adding a default parameter or just force the callsites to create an instance. There should be very few callsites anyway. It doesn't seem necessary to introduce two ways to construct the resolver.

Reply:
There are 2 parameters exposed in the new ctor.  `ReaderParameters` is easy enough to force on the callsite.  `extensions` is the awkward one.  I could say if `extensions` is null, use  `Extensions`, but that's a bit ugly.  Or I could add `.winmd` to `Extensions` and avoid exposing `extensions` in the ctor.

`public AssemblyResolver (LinkContext context)` is public so I was leaning toward avoiding an API break in case anything outside of the linker is using it.

I'm happy to change it to whatever you'd like it to be.  Let me know what you'd prefer @vitek-karas .

Reply:
I think adding `windm` to the list of extensions is a good solution to the extensions problem (just like we're adding it elsewhere).

`AssemblyResolver` is declared as public - but it's not part of the ref package, so we might as well change it to internal (which might not work for you though). So I'm not worried about breaking anything - and if it does - good, I want to find that and remove that dependency as soon as possible :-)

Reply:
Done.